File: nivel-05/ex-01/somaPoo.c
// Código que retorna o dobro de um numero utilizando POO

#include <iostream>
#include <assert.h>

class Calc {
public:
    int dobro(int num) {
    	return num * 2;
    }
};

void testDobro() {
    Calc c;

    assert(10 == c.dobro(5));
    assert(20 == c.dobro(10));
}

int main() {
    testDobro();
    
}

File: nivel-05/ex-01/soma_poo.py
# soma dois numeros usando OO

class Calc(object):
	'Uma simples classe'
	def __init__(self, num):
		self.num = num
	def dobro(self):
		return self.num*2

'''Instanciando classes'''
c = Calc(2) 
d = Calc(3)

'''Testando classes'''
assert  c.dobro() == 4
assert  d.dobro() == 6